cogs/github.py

```diff
@@ -0,0 +1,101 @@
+from discord import Embed, Color
+from discord.ext.commands import Cog, Context, command, Bot, cooldown, BucketType
+from .utils.constants import Emojis
+import aiohttp
+
+BAD_RESPONSES = {
+    404: "Issue/pull request not Found! Please enter a valid PR Number!",
+    403: "Rate limit is hit! Please try again later!"
+}
+
+
+# TODO : add gitlab commands too ig.
+class Github(Cog):
+    def __init__(self, bot: Bot) -> None:
+        self.bot = bot
+        self.session = aiohttp.ClientSession()
+
+    @staticmethod
+    def generate_description(description: str, stars: str, forks: str, command: str) -> str:
+        return f"**{description}**\nStars: {stars} Forks: {forks}\n Command: {command}"
+
+    @command()
+    @cooldown(1, 5, type=BucketType.user)
+    async def issue(self, ctx: Context, number: int, repository: str = "HotWired-Bot", user: str = "The-Codin-Hole") -> None:
+        """Command to retrieve issues from a GitHub repository."""
+        url = f"https://api.github.com/repos/{user}/{repository}/issues/{number}"
+        merge_url = f"https://api.github.com/repos/{user}/{repository}/pulls/{number}/merge"
+
+        async with self.session.get(url) as resp:
+            json_data = await resp.json()
+
+        if resp.status in BAD_RESPONSES:
+            await ctx.send(f"ERROR CODE : [{str(resp.status)}] {BAD_RESPONSES.get(resp.status)}")
```

Comment:
The user doesn't need to see the actuall error code
```suggestion
            await ctx.send(f"ERROR: [ {BAD_RESPONSES.get(resp.status)}")
```